src/spec/gocli/integration/cli_stemcell_spec.rb

```diff
@@ -32,13 +32,18 @@
   end
 
   context 'if cpi config is used' do
-    it 'creates a stemcell for each configured cpi' do
+    it 'creates a stemcell for each configured cpi if it matches stemcell format' do
       stemcell_filename = spec_asset('valid_stemcell.tgz')
 
       cpi_path = current_sandbox.sandbox_path(Bosh::Dev::Sandbox::Main::EXTERNAL_CPI)
       cpi_config_manifest = yaml_file('cpi_manifest', Bosh::Spec::Deployments.simple_cpi_config(cpi_path))
       bosh_runner.run("update-cpi-config #{cpi_config_manifest.path}")
 
+      stemcell_filename_not_dummy = spec_asset('valid_stemcell_not_dummy_stemcell_format.tgz')
```

Comment:
These new assertions should be in a different `it` block. This test currently mixes valid stemcells and invalid stemcells, so the expected behavior is testing two different things. There should be a new test which tests the unsupported stemcells explicitly.

So there should be one test which successfully uploads a stemcell and we can see the stemcells in the stemcell table, and a new test which uploads a stemcell, with the wrong format, and the stemcells table should not have this new stemcell.